### PR TITLE
Fix allowed python versions of py-tables

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-tables/package.py [hunk 1]
```diff
@@ -29,8 +29,8 @@ class PyTables(PythonPackage):
     variant('lzo', default=False, description='Support for lzo compression')
 
     # requirements.txt
-    depends_on('python@3.5:', when='@3.4:', type=('build', 'run'))
-    depends_on('python@2.6:', type=('build', 'run'))
+    depends_on('python@2.7:2.7.999,3.5:', when='@3.4:', type=('build', 'run'))
+    depends_on('python@2.6:2.6.999',  when='@:2',   type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-tables/package.py]: These versions aren't in Spack

Comment (iarspider) [reply var/spack/repos/builtin/packages/py-tables/package.py]: I was trying to preserve the original version ranges as much as possible. Cleaned up.

File: var/spack/repos/builtin/packages/py-tables/package.py [hunk 2]
```diff
@@ -29,8 +29,7 @@ class PyTables(PythonPackage):
     variant('lzo', default=False, description='Support for lzo compression')
 
     # requirements.txt
-    depends_on('python@3.5:', when='@3.4:', type=('build', 'run'))
-    depends_on('python@2.6:', type=('build', 'run'))
+    depends_on('python@2.7:2.7.999,3.0:', when='@3.2.2:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-tables/package.py]: This change doesn't make any sense. Before, you could build older versions with Python 2.6+ and newer versions with Python 3.5+. Now all versions can be built with Python 2.7+, which is not the case.

**Comment by adamjstewart** [comment]
@iarspider is this still a work-in-progress or can we close this PR?

File: var/spack/repos/builtin/packages/py-tables/package.py [hunk 3]
```diff
@@ -29,7 +29,7 @@ class PyTables(PythonPackage):
     variant('lzo', default=False, description='Support for lzo compression')
 
     # requirements.txt
-    depends_on('python@3.5:', when='@3.4:', type=('build', 'run'))
+    depends_on('python@2.6:2.8,3.5:', when='@3.4:', type=('build', 'run'))
     depends_on('python@2.6:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-tables/package.py]: I believe the correct ranges are:
```python
depends_on('python@3.5:', when='@3.6.1:', type=('build', 'run'))
depends_on('python@3.4:', when='@3.6.0:', type=('build', 'run'))
depends_on('python@2.6:', type=('build', 'run'))
```

**Comment by adamjstewart** [comment]
var/spack/repos/builtin/packages/py-tables/package.py:35: [W293] blank line contains whitespace

**Pull request merged**